src/Interpreters/Cache/FileCache_fwd.h

```diff
@@ -4,7 +4,7 @@
 namespace DB
 {
 
-static constexpr int FILECACHE_DEFAULT_MAX_FILE_SEGMENT_SIZE = 100 * 1024 * 1024;
+static constexpr int FILECACHE_DEFAULT_MAX_FILE_SEGMENT_SIZE = 8 * 1024 * 1024;
```

Comment:
isn't it too small?

Reply:
first of all, why change it. since we start to share segments between tasks ('cause of alignment) now some of them that actually need to read only from the end of the segment could read the whole segment in pre-download. so it seems logical to limit size of the segments.
from what perspective do you think it might be too small? overhead per segment during processing seems very small. the only downside I could think of is prob number of GetObject-s will increase.

Reply:
I also saw this effect, in my case it was because of connection timeouts that start to happen more frequently when we increase number of parallel requests. for this i added conn pool for s3 reading in the second pr.
and I like your idea of coalescing reads on GetObject level